contributors/design-proposals/priority-eviction.md

```diff
@@ -0,0 +1,60 @@
+# Introducing Priority to Kubelet Memory Eviction
+
+**Author**: David Ashpole (@dashpole)
+
+**Last Updated**: 7/25/2017
+
+**Status**: Proposal
+
+This document explores various schemes to include priority in kubelet memory evictions
+
+## Introduction
+
+### Definitions
+["Priority"](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/pod-priority-api.md) is an integer, somehow set by users when running their pods.  Assumed to be intentionally set, and controlled properly by mechanisms outside of this proposal.  
+["Quality of Service"](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-qos.md), or QoS, is the performance SLO kubernetes provides based on their resource requests and limits.
+ - Guaranteed: Requests == Limits  
+ - Burstable: Requests < Limits  
+ - Besteffort: No Requests  
+
+["Memory Eviction"](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/kubelet-eviction.md): is the process of removing a pod from the kubelet when under memory pressure in order to free up resources.  Eviction decisions are made at the node level by the kubelet.  
+"Preemption": is the process of deleting one pod from a node in order to make room for another pod, which is deemed by the scheduler to be more important to run.  Preemption decisions are made at the cluster level by the scheduler.  
+
+### Background and Motivation
+Prior to kubernetes v1.6, the [critical pod](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/rescheduling-for-critical-pods.md) annotation was introduced primarily to prevent eviction of system pods deemed critical.  In some cluster setups, a prerequisite for moving away from using this annotation is a properly designed system of priority that decreases the chance that high-priority pods are evicted, and is able to properly reschedule such pods quickly when they are evicted.  The [Kubernetes Priority](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/pod-priority-api.md) proposal introduced this concept of priority to kubernetes v1.7.
+
+This proposal explores possible implementations for integrating priority with the kubelet process of eviction.  This integration should provide a method for cluster operators to provide greater availability to "high-priority" pods without needing to use the critical-pod annotation.  
+
+The current method for ranking pods for eviction is by QoS, then usage over requests.  This currently holds the invariant that a pod that does not exceed its requests is not evicted, since the sum of the requests on the node cannot exceed the allocatable memory on the node.
```

Comment:
for clarity, is oom_score_adj behavior worth explaining here as i think they go hand-in-hand.